src/tasks/shop-admin/Product/BulkEditProducts.ts

```diff
@@ -0,0 +1,123 @@
+                    if (changes['restockTime'].method != 'Clear') {
+                        await AdminProductBulkEdit.restockTimeInput.fill(changes['restockTime'].value);
+                    } 
+                }
+
+                if (changes['tags'] != null) {
+                    await AdminProductBulkEdit.changeTagsCheckbox.click();
+                    await AdminProductBulkEdit.tagsChangeMethodDropdown.click();
+                    await (await AdminProductBulkEdit.getDropdownEntry(changes['tags'].method)).click();
+                    if (changes['tags'].method != 'Clear') {
+                        await AdminProductBulkEdit.tagsInput.click();
+                        const responsePromise = AdminProductBulkEdit.page.waitForResponse(response =>
+                            response.url().includes(`search/tag`) && response.status() === 200 && response.request().method() === 'POST'
+                        );
+                        await AdminProductBulkEdit.tagsInput.fill(changes['tags'].value);
+
+                        // Wait for the search call to be completed
+                        const response = await responsePromise;
+                        expect(response.ok()).toBeTruthy();
+                        const searchResult = await AdminProductBulkEdit.getDropdownEntry(changes['tags'].value);
+                        await searchResult.click();
+                    } 
+                }
+
+                if (changes['saleschannel'] != null) {
+                    await AdminProductBulkEdit.changeSalesChannelCheckbox.click();
+                    await AdminProductBulkEdit.salesChannelChangeMethodDropdown.click();
+                    await (await AdminProductBulkEdit.getDropdownEntry(changes['saleschannel'].method)).click();
+                    if (changes['saleschannel'].method != 'Clear') {
+                        await AdminProductBulkEdit.salesChannelInput.click();
+                        const responsePromise = AdminProductBulkEdit.page.waitForResponse(response =>
+                            response.url().includes(`search/sales-channel`) && response.status() === 200 && response.request().method() === 'POST'
+                        );
+                        await AdminProductBulkEdit.salesChannelInput.fill(changes['saleschannel'].value);
+
+                        // Wait for the search call to be completed
+                        const response = await responsePromise;
+                        expect(response.ok()).toBeTruthy();
+                        const searchResult = await AdminProductBulkEdit.getDropdownEntry(changes['saleschannel'].value);
+                        await searchResult.click();
+                    } 
+                }
+
+                await AdminProductBulkEdit.applyChangesButton.click();
+                await AdminProductBulkEdit.confirmModalApplyChangesButton.click();
+                await expect(AdminProductBulkEdit.confirmModalSuccessHeader).toBeVisible();
```

Comment:
Between that both command there will be a operation...should we check properly if the operation is done? Maybe the current implementation could lead to flakyness...

Reply:
Added a check for the loading spinner to appear and disappear again. I think we can't wait for api calls, because we don't know at this point, which calls are fired. There are multiple depending on which fields are changed